### PR TITLE
Email Input

### LINKED ISSUE #4947: create a special email input that checks if there is an existing invitation for this email
create a special email input that checks if there is an existing invitation for this email.

If we find a pending invitation from org to joinOrg or to AttendEvent => 
- display preGeneratedPassord field
- if invitation is `joinOrg` display org information from algolia

Maybe use or update `hasUserAnOrgOrIsAlreadyInvited` ?

### PR DESCRIPTION
**Pull request by Mussuranae**
New email component for the new connection flow
Fix #4947

### CODE REVIEW

File: apps/backend-functions/src/invitation.ts [hunk 1]
```diff
@@ -211,3 +211,24 @@ export async function hasUserAnOrgOrIsAlreadyInvited(userEmails: string[]) {
 
   return invitationQuery.some(d => d.docs.length > 0);
 }
+
+export async function getInvitationLinkedToEmail(email: string) {
```
Comment (phpgeekfr) [inline apps/backend-functions/src/invitation.ts:215]: can you type the returned value ? `promise<boolean|publicOrg>`

File: libs/user/src/lib/auth/components/email-input/email-input.component.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  templateUrl: './email-input.component.html',
+  styleUrls: ['./email-input.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class EmailInputComponent implements OnInit {
+
+  @Output() private readonly organization = new EventEmitter<PublicOrganization>();
+  @Output() showInvitationInput = new EventEmitter<boolean>(false);
+
+  private getInvitationLinkedToEmail = this.functions.httpsCallable('getInvitationLinkedToEmail')
```
Comment (phpgeekfr) [inline libs/user/src/lib/auth/components/email-input/email-input.component.ts:17]: maybe more clean to put this into invitationService ?

**Review by phpgeekfr — APPROVED** [review]
LGTM

File: libs/user/src/lib/auth/components/email-input/email-input.component.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  public emailForm = new FormControl();
+
+  constructor(private functions: AngularFireFunctions) { }
+
+  ngOnInit() {
+    this.emailForm.valueChanges.subscribe(_ => this.searchForInvitation());
```
Comment (phpgeekfr) [inline libs/user/src/lib/auth/components/email-input/email-input.component.ts:23]: can you make a call to function only if we have a complete email ?

Comment (Mussuranae) [reply libs/user/src/lib/auth/components/email-input/email-input.component.ts]: Done with the `email` attribute in the html.

**Pull request merged**